internal/pkg/scaffold/cmd.go

```diff
@@ -159,9 +159,19 @@ func main() {
 		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
 	}
 	// Create Service object to expose the metrics port(s).
-	_, err = metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
 	if err != nil {
-		log.Info(err.Error())
+		log.Info("Could not create metrics Service", "error:", err.Error())
+	}
+
+	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
+	// necessary to configure Prometheus to scrape metrics from this operator.
+	// If this operator is deployed to a cluster without the prometheus-operator running, it will return
+	// ErrServiceMonitorNotPresent,which can be used to safely skip ServiceMonitor creation.
+	services := []*v1.Service{service}
+	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	if err != nil {
```

Comment:
Can we just implement the ErrServiceMonitorNoPresent check to skip the return of error so this can work out of the box on vanilla k8s?

Reply:
Sure, but this will work out of the box already as we have a check in place that makes sure ServiceMonitor is present, but can add the error check and log the error.